driver.sh

```diff
@@ -0,0 +1,82 @@
+set -ux
+
+check_dependencies() {
+  set -e
+
+  test -x `which cmake`
+  test -x `which ninja`
+  test -x `which gcc`
+  test -x `which aarch64-linux-gnu-as`
+  test -x `which aarch64-linux-gnu-ld`
+  test -x `which qemu-system-aarch64`
+  test -x `which timeout`
+  test -x `which unbuffer`
+
+  set +e
+}
+
+build_clang() {
+  ln -sf ../../clang llvm/tools/clang
+  mkdir -p llvm/build
+  cd llvm/build
+
+  cmake -G Ninja -DCMAKE_BUILD_TYPE=Release ..
```

Comment:
Might want to look at some of the threading options like `LLVM_PARALLEL_COMPILE_JOBS` and `LLVM_PARALLEL_LINK_JOBS` since Travis timed out (I didn't see an error so I just assume it wasn't going fast enough but don't have time to read [this](https://docs.travis-ci.com/user/common-build-problems/#my-builds-are-timing-out)). Can also probably ccache it up (`LLVM_CCACHE_BUILD`)

Reply:
Talking with some teammates, their ideas were to disable other backends that we don't need.  I consider that an antipattern for the distribution of clang, but to speed up our tests, it may be fine.

Travis also had some documentation I recall on using ccache, then caching that cache dir between builds.

Oh, looks like there are nightly builds of clang? https://apt.llvm.org/  That would save a lot of time!

Reply:
I think switching to the nightly builds is ideal so that we don't have to tax Travis lol.